src/mjolnir/graphbuilder.cc

```diff
@@ -138,7 +138,8 @@ void ConstructEdges(const OSMData& osmdata,
     auto way_node = *way_nodes[current_way_node_index];
     const auto way = *ways[way_node.way_index];
     const auto first_way_node_index = current_way_node_index;
-    const auto last_way_node_index = first_way_node_index + way.node_count() - 1;
+    const auto last_way_node_index =
+        first_way_node_index + way.node_count() - way_node.way_shape_node_index - 1;
```

Comment:
because we can visit a way twice now (when it doubles back) we need to worry about not starting at the 0th node of a way